### PR TITLE
feat(a380x/fcu): Fix visual issues with FCU light test and missing labels

### LINKED ISSUE #9091: A380X/FCU: Light test speed window overlap
### Aircraft Version

Development

### Build info

```json
master
```

### Describe the bug

The "KT" text overlaps with the "8" on the speed window when performing a light test.

### Expected behavior

Shouldn't overlap

### Steps to reproduce

Perform a light test.

### References (optional)

image

### Additional info (optional)

_No response_

### Discord Username (optional)

bruno_pt99

### PR DESCRIPTION
**Pull request by heclak**
Fixes #9091

## Summary of Changes

* Update display values on the FCU for the light test to match the reference image
* Change decimal point width in FCU font to match font and display in reference. This fixes the overlapping text issue in the speed display. This replaces an older PR #9129
* Move the FPA label to the left of the V/S label as per reference
* Add TRUE label. The label is not yet connected to TRUE mode but appears in the light test. Will be connected in a separate PR
* Move the degree symbol closer to the speed value to match the reference
* Changed first digit of FPA/VS to include different test digit (includes a +- symbol within the 8)
* Change FPA to not use the left most +- character. FPA +- will be aligned under the "F" character
* Add test pattern for EFIS CP labels

## Screenshots (if necessary)

Old image
image
image

Updated Images
image
FlightSimulator_2024-11-23-10-41-55-260

FPA Symbol
image

## References

https://www.flickr.com/photos/75877584@N05/49491034416/in/photolist-2ipmuhQ-siX6c1-syfays-sAn7SW/

image
https://www.airliners.net/photo/Lufthansa/Airbus-A380-841/2736833/L?qsp=eJwtjEEKwjAQRa8is3ZRKYh0Vz2ALrzAMPloaDVhMqCh9O5Og7vH%2B5%2B3kKS34Wv3mkEDFbDKk/aUWflVaFhoQv0kDc409qdud0ky5Wj%2BKUntXN0HNowiyIbw91cN0G1CkZZ7eP7gAL01pv7oPsSSZ24NGMeZ1vUHSF4wZA%3D%3D

image

## Additional context

Discord username (if different from GitHub): Heclak

## Testing instructions

1. Set ANN LT sw to TEST. Check indication on FCU matches references and updated screenshots.
2. Switch modes on the FCU to check that labels are displaying/hiding correctly outside of TEST mode. Change numbers on FCU.
3. Check that preselect baro setting functions correctly. Preselect should timeout after a short time when in STD mode. 

## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo**, **flybywire-aircraft-a380-842 (4K)** or **flybywire-aircraft-a380-842 (8K)** download link at the bottom of the page

### CODE REVIEW

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -69,6 +69,7 @@
 1. [FMS] Fix Pause at T/D not working in selected speed - @BlueberryKing (BlueberryKing)
 1. [MISC] Replaced brake temperature simulation with physics based model of brakes - @Gurgel100 (Pascal)
 1. [A32NX/MCDU] Suppress TMPY FPLN when no modifications made in airways page - @robertxing2004 (robeet)
+1. [A380X/FCU] Fix display of values on FCU during light test - @heclak (Heclak)
```
Comment (tracernz) [inline .github/CHANGELOG.md:72]: Move to the end of the list please.

**Review by tracernz — COMMENTED** [review]
The true mode is already working FYI, so could be hooked up.

File: .github/CHANGELOG.md (hunk 1 above)
Comment (G-DRAW) [reply .github/CHANGELOG.md:72]: This is fantastic, well done FBW!

Comment (heclak) [reply .github/CHANGELOG.md]: @tracernz moved.

File: .github/CHANGELOG.md [hunk 2]
```diff
@@ -69,9 +69,10 @@
 1. [FMS] Fix Pause at T/D not working in selected speed - @BlueberryKing (BlueberryKing)
 1. [MISC] Replaced brake temperature simulation with physics based model of brakes - @Gurgel100 (Pascal)
 1. [A32NX/MCDU] Suppress TMPY FPLN when no modifications made in airways page - @robertxing2004 (robeet)
-1. [A380X/FMC] Fix reset of perf data on done phase or database swap - @tracernz (Mike)
-1. [A32NX/ATC] Fixed ATC/TCAS power supply - @tracernz (Mike)
-1. [A32NX] Fixed appearance of FCU decals on MSFS2024 - @tracernz (Mike)
+2. [A380X/FMC] Fix reset of perf data on done phase or database swap - @tracernz (Mike)
+3. [A32NX/ATC] Fixed ATC/TCAS power supply - @tracernz (Mike)
+4. [A32NX] Fixed appearance of FCU decals on MSFS2024 - @tracernz (Mike)
+5. [A380X/FCU] Fix display of values on FCU during light test - @heclak (Heclak)
```
Comment (2hwk) [inline .github/CHANGELOG.md:75]: Diff has been messed up here, change to all `1.`

Comment (heclak) [reply .github/CHANGELOG.md]: ah sorry. didn't realise vs code has formatted it. cleaned up.

**Review by tracernz — APPROVED** [review]

**Comment by heclak** [comment]
FCU updates are complete for this PR. TRUE indication will be done in another PR due to some refactoring of the heading indication that will be done at the same time.

File: fbw-a380x/src/systems/instruments/src/FCU/Components/NdData.tsx [hunk 3]
```diff
@@ -33,6 +42,15 @@ export class NdData extends DisplayComponent<NdDataProps> {
...
+
+  onAfterRender(_node: VNode): void {
+    this.sub
+      .on('ovhd_ann_lt_test_active')
+      .whenChanged()
+      .handle((v) => this.lightsTest.set(v));
+  }
+
   render(): VNode | null {
     return (
```
Comment (tracernz) [inline fbw-a380x/src/systems/instruments/src/FCU/Components/NdData.tsx:55]: I would recommend just using the same method as the navaid images for all of these.

Comment (heclak) [reply fbw-a380x/src/systems/instruments/src/FCU/Components/NdData.tsx:55]: so using the map function? I'll give it a try again. I tried a few different methods and had trouble getting it to work but I think understand things a bit better now.

File: fbw-a380x/src/systems/instruments/src/FCU/Components/NdData.tsx [hunk 4]
```diff
@@ -42,27 +60,131 @@ export class NdData extends DisplayComponent<NdDataProps> {
...
+            style="position: absolute; top: 0; left: 2790px"
+            width="620px"
+            class={{ hidden: this.lightsTest }}
+            src="/Images/fbw-a380x/fcu/TRAF.png"
+          />
           <img
             style="position: absolute; top: 0; left: 3740px"
             width="620px"
+            class={{ hidden: this.lightsTest }}
             src={this.navaidMode2.map((v) => NdData.NAVAID_2_IMAGES[v])}
```
Comment (tracernz) [inline fbw-a380x/src/systems/instruments/src/FCU/Components/NdData.tsx:155]: Yeah, like this. For the navaids specifically you can just add an extra state for light test.

**Comment by heclak** [comment]
Are there any changes needed to move this to QA testing? Would prefer to let this merge before doing some additional refactoring of the FCU code.

**Comment by BravoMike99** [comment]
Quality Assurance Tester Report

Discord Username  : bruno_pt99
Object of testing : #9570  
Aircraft          : A380X 
Tier of Testing   : 1
Date              : 07/12/2024 

Testing Process: 
1) Turned on the light test and verified the indications on the FCU. They match screenshots & refs ✔️ 
 2) Tested TRK/FPA, MACH/SPD & TRUE/MAG & changed some FCU values. All labels were updated on the FCU ✔️ .
 3) Tested barometric preselection and it worked ✔️ 
 

Testing Results: 
Passed  

Media:
Light test

**Pull request merged**